app/scripts/lib/ipfsContent.js

```diff
@@ -0,0 +1,36 @@
+const extension = require('extensionizer')
+const resolver = require('./resolver.js')
+
+module.exports = function (provider) {
+    extension.webRequest.onBeforeRequest.addListener(details => {
+        const name = details.url.substring(7, details.url.length - 1)
+        let clearTime = null
+        extension.tabs.getSelected(null, tab => {
+            extension.tabs.update(tab.id, { url: 'loading.html' })
+
+            clearTime = setTimeout(() => {
+                return extension.tabs.update(tab.id, { url: '404.html' })
+            }, 60000)
+
+            resolver.resolve(name, provider).then(ipfsHash => {
+                clearTimeout(clearTime)
+                let url = 'https://gateway.ipfs.io/ipfs/' + ipfsHash
```

Comment:
Are we sure we want to hit IPFS gateway and not Infura?

Sending this comment to Infura for comment.

Reply:
Sure, I have changed to the `ipfs.infura.io`